src/components/Swap/Toolbar/index.tsx

```diff
@@ -37,6 +37,28 @@ const ToolbarRow = styled(Row)`
   padding: 0 1em;
 `
 
+const Context = createContext<{
+  open: boolean
+  collapse: () => void
+  toggleOpen: () => void
+}>({
+  open: false,
+  collapse: () => null,
+  toggleOpen: () => null,
+})
+
+export const Provider = ({ children }: PropsWithChildren) => {
+  const [open, setOpen] = useState(false)
+  const toggleOpen = useCallback(() => setOpen((open) => !open), [])
+  const collapse = useCallback(() => setOpen(false), [])
```

Comment:
dont need to wrap these in `useCallbacks` since theyre just simple function definitions :P